src/tig.c

```diff
@@ -475,10 +475,22 @@ parse_options(int argc, const char *argv[], bool pager_mode)
 
 	request = pager_mode ? REQ_VIEW_PAGER : REQ_VIEW_MAIN;
 
-	if (argc <= 1)
+	/* Options that must come before any subcommand. */
+	for (i = 1; i < argc; i++) {
+		const char *opt = argv[i];
+		if (!strncmp(opt, "-C", 2)) {
+			if (chdir(opt + 2))
+				die("Failed to change directory to %s", opt + 2);
+			continue;
+		} else {
+			break;
+		}
+	}
+
+	if (i >= argc)
 		return request;
 
-	subcommand = argv[1];
+	subcommand = argv[i++];
```

Comment:
The subcommand is optional so the `++` here is what I think is causing problems.

```c
int i;

//...
for (i = 1; i < argc; i++) {
        // parse pre-arg: -C
}

subcommand = argv[i];
// parse subcommand

if (subcommand)
        i++;

for (; i < argc; i++) {
        // parse remaining args
}
```

Reply:
Yep, I think our comments crossed. Should be fixed by the commit I just pushed up.